src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -499,6 +507,12 @@ public boolean isEnableNewJsonParsingLogic() {
     return (val instanceof String) ? Boolean.parseBoolean(val.toString()) : (boolean) val;
   }
 
+  /** @return Whether the client is in Iceberg mode */
+  public boolean isIcebergMode() {
+    Object val = this.parameterMap.getOrDefault(STREAMING_ICEBERG, STREAMING_ICEBERG_DEFAULT);
+    return (val instanceof String) ? Boolean.parseBoolean(val.toString()) : (boolean) val;
```

Comment:
The parameter map should be storing validated and parsed values instead of doing it at runtime. This method is called on every row insert and probably multiple times. Such a waste of CPU to do this parseBoolean thosands of times a second!

Reply:
Added cached value and custom exception for this.